enacts/enactstozarr.py

```diff
@@ -48,53 +49,27 @@ def set_up_dims(xda):
     # and look into xESMF.
     #
     # [1] https://climatedataguide.ucar.edu/climate-data-tools-and-analysis/regridding-overview
-    rr_mrg = rr_mrg.interp(
-        X=np.arange(rr_mrg.X.min(), rr_mrg.X.max() + RESOLUTION, RESOLUTION),
-        Y=np.arange(rr_mrg.Y.min(), rr_mrg.Y.max() + RESOLUTION, RESOLUTION),
-    )
-
-rr_mrg = rr_mrg.chunk(chunks=CHUNKS)
-
-xr.Dataset().merge(rr_mrg).to_zarr(
-  store = RR_MRG_ZARR_PATH
-)
-
-if not os.access(CONFIG["data_dir"], os.W_OK | os.X_OK):
-    sys.exit("can't write to output directory")
-
-def set_up_dims(xda):
-    datestr = Path(xda.encoding["source"]).name.split("_")[2]
-    year = int(datestr[0:4])
-    month = int(datestr[4:6])
-    dekad = int(datestr[6:7])
-    day = (dekad - 1) * 10 + 1
-    xda = xda.expand_dims(T = [dt.datetime(year, month, day)])
-    xda = xda.rename({'Lon': 'X','Lat': 'Y'})
-    return xda
-
-def convert(variable):
-    netcdf = list(sorted(Path(CONFIG['data_src'][variable]).glob("*.nc")))
-
-    data = xr.open_mfdataset(
-        netcdf,
-        preprocess = set_up_dims,
-        parallel=False
-    )#[variable]
-    print(data)
-
+        data = data.interp(
+            X=np.arange(data.X.min(), data.X.max() + ZARR_RESOLUTION, ZARR_RESOLUTION),
+            Y=np.arange(data.Y.min(),data.Y.max() + ZARR_RESOLUTION, ZARR_RESOLUTION),
+        )    
+    
     data = data.chunk(chunks=CONFIG['chunks'])
-
-    zarr = f"{CONFIG['data_dir']}/{variable}.zarr"
+    
+    zarr = f"{CONFIG['zarr_path']}{CONFIG['vars'][variable][1]}"
+    
     shutil.rmtree(zarr, ignore_errors=True)
     os.mkdir(zarr)
-
+    
     xr.Dataset().merge(data).to_zarr(
         store = zarr
     )
-
+    
+    if not os.access(zarr, os.W_OK | os.X_OK):
+        sys.exit("can't write to output directory")
+    
+    print(f"conversion for {variable} complete.")
     return zarr
 
-
-convert("rfe")
-convert("tmin")
-convert("tmax")
+for i in CONFIG['vars']:
+    convert(i)
```

Comment:
I have something in progress that will let us incrementally add data to a zarr store as the upstream dataset gets extended in time, rather than re-converting the entire dataset on every run of the script. Once that's available, we'll rewrite this script in that paradigm; in the meantime, don't worry about it.

Reply:
Yes and I guess my thought was someone could just comment out variables they wouldn't want for now.